### PR TITLE
Replace MessDetector/CamelCasePropertyName.php and rulesets/controversial.xml/Superglobals by Sniffs

### PR DESCRIPTION
**Pull request by TomasVotruba**
Follow up to #6

Tested on Shopsys framework `/src`: https://git.shopsys-framework.com/

---

###  `_` prefix of private properties was fixed

`PSR2.Classes.PropertyDeclaration.Underscore`  spots them:

```bash
FILE: ...var/www/coding-standards/src/Sniff/ForbiddenSuperGlobalSniff.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 17 | WARNING | Property name "$_superglobalVariables" should not be
    |         | prefixed with an underscore to indicate visibility
    |         | (PSR2.Classes.PropertyDeclaration.Underscore)
----------------------------------------------------------------------
```

with proper configuration:

```xml
<rule ref="Squiz.NamingConventions.ValidVariableName">
    <!-- private properties should not start with "_" -->
    <exclude name="Squiz.NamingConventions.ValidVariableName.PrivateNoUnderscore"/>
</rule>
```

### Related Changes

- bump to PHP 7.0, as discussed with @vitek-rostislav and @PetrHeinz 
- add PHP 7.0 strict_types and typehints

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Ready to review and merge :+1:

File: src/Sniff/ForbiddenSuperGlobalSniff.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+final class ForbiddenSuperGlobalSniff implements Sniff
+{
+    /**
+     * @var string[]
+     */
+    private $superglobalVariables = ['$_COOKIE', '$_GET', '$_FILES', '$_POST', '$_REQUEST', '$_SERVER'];
+
+    /**
+     * @return int[]
```
Comment (vitek-rostislav) [inline src/Sniff/ForbiddenSuperGlobalSniff.php]: Why do not you use the PHP7 typehint here?

File: src/Sniff/ForbiddenSuperGlobalSniff.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    public function register()
+    {
+        return [T_VARIABLE];
+    }
+
+    /**
+     * @param int $position
```
Comment (vitek-rostislav) [inline src/Sniff/ForbiddenSuperGlobalSniff.php]: missing annotation for $file parameter

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
Hello @TomasVotruba. 

thank you for your PR, I added two simple comments, review them please.

File: src/Sniff/ForbiddenSuperGlobalSniff.php (hunk 1 above)
Comment (TomasVotruba) [reply src/Sniff/ForbiddenSuperGlobalSniff.php]: I missed that, nice catch

File: src/Sniff/ForbiddenSuperGlobalSniff.php (hunk 2 above)
Comment (TomasVotruba) [reply src/Sniff/ForbiddenSuperGlobalSniff.php]: The `File` type is actually in the argument already

**Comment by TomasVotruba** [comment]
Thanks you @vitek-rostislav , I've added the return typehint

**Comment by TomasVotruba** [comment]
Build failed due to stuck Travis, please "rerun" it here: https://travis-ci.org/shopsys/coding-standards/jobs/325052223

File: src/Sniff/ForbiddenSuperGlobalSniff.php (hunk 2 above)
Comment (vitek-rostislav) [reply src/Sniff/ForbiddenSuperGlobalSniff.php]: Ok, thanks for clarification.

**Review by vitek-rostislav — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you!

**Comment by vitek-rostislav** [comment]
Thank you! :)

**Comment by TomasVotruba** [comment]
Oh, I forgot that

Thank you sir!

**Pull request merged**